### PR TITLE
Make scene.save() support saving pdf, ps, and eps files.

### PR DESCRIPTION
**Pull request by ngoldbaum**
This makes it so you can save pdf, eps, and ps files directly from the scene object, making use of matplotlib to handle the file format conversion issues. I made sure that if you save an image as a pdf and a png, the images you get back match orientation and appearance as closely as possible.

This is technically a new feature but it was simple enough to add that I went ahead and did it. Given the bug report (#1503) and the vagueness of the current docstrings I don't think it's crazy to expect yt to do the right thing if you tell it to save a pdf instead of a png.

~~Let me know if my use of pyplot is objectionable - it was easier to implement this using pyplot rather than using the OO interface. The pyplot import only happens if you tell yt to save something that's not a png so I think it should be safe for existing scripts. I could probably rewrite it to use the matplotlib OO interface if anyone sees using pyplot as a possible issue.~~

EDIT: It looks like travis doesn't like using pyplot without a DISPLAY variable set. That was enough to kick me in the pants and remove the pyplot usage.

### CODE REVIEW

File: yt/visualization/volume_rendering/scene.py [hunk 1]
```diff
@@ -302,7 +304,28 @@ def save(self, fname=None, sigma_clip=None):
...
+            fig.subplots_adjust(left=0, bottom=0, right=1, top=1)
+            ax.set_axis_off()
+            out = self._last_render
+            nz = out[:, :, :3][out[:, :, :3].nonzero()]
+            max_val = nz.mean() + sigma_clip * nz.std()
+            alpha = 255 * out[:, :, 3].astype('uint8')
+            out = np.clip(out[:, :, :3] / max_val, 0.0, 1.0) * 255
+            out = np.concatenate(
+                [out.astype('uint8'), alpha[..., None]], axis=-1)
+            # not sure why we need rot90, but this makes the orentation
```
Comment (zingale) [inline yt/visualization/volume_rendering/scene.py]: just to be sure -- it is a rotate and not a transpose that is needed?

Comment (ngoldbaum) [inline yt/visualization/volume_rendering/scene.py]: Yup. For testing I was using this script:

```
import yt

ds = yt.load('IsolatedGalaxy/galaxy0030/galaxy0030')

sc = yt.create_scene(ds, 'density', lens_type='perspective')

for ext in ['.pdf', '.png']:
    sc.save('rendering{}'.format(ext), sigma_clip=6.0)
```

Which produced the following two files:

rendering
[rendering.pdf](https://github.com/yt-project/yt/files/1170675/rendering.pdf)

**Comment by luminosa42** [comment]
I tested it, looks good! Thanks!

**Comment by ngoldbaum** [comment]
Thanks for taking a look and for taking the time to report the issue!

**Pull request merged**